scripts/apply-ksappend.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/python3
+
+import sys
+import os
+import argparse
+import shutil
+from pykickstart import parser as kickstart_parser
+
+FRAGMENTS_FOLDER = "fragments"
+DYNAMIC_FOLDER = os.path.join(FRAGMENTS_FOLDER, "dynamic")
+TOPLEVEL_PLATFORM_FOLDER = os.path.join(FRAGMENTS_FOLDER, "platform")
+
+
+class ArgumentParser(object):
+
+    def __init__(self):
+        super().__init__()
+        self._parser = argparse.ArgumentParser(description="""
+        Include fragments linked by %ksappend in tests.
+        """)
+
+        self._test_file_path = ""
+        self._platform_name = ""
+
+        self._configure_parser()
+
+    @property
+    def test_file_path(self):
+        return self._test_file_path
+
+    @property
+    def platform_name(self):
+        return self._platform_name
+
+    @property
+    def platform_dir_path(self):
+        return os.path.join(TOPLEVEL_PLATFORM_FOLDER, self._platform_name)
+
+    def _configure_parser(self):
+        self._parser.add_argument("--test_file_path", "-t", required=True, type=str,
+                                  help="Path to the kickstart test file to process.")
+        self._parser.add_argument("--platform_name", "-p", type=str, required=True,
+                                  help="Name of the platform folder for dynamic ksappend fragments.")
+
+    def parse(self):
+        ns = self._parser.parse_args()
+
+        self._test_file_path = ns.test_file_path
+        self._platform_name = ns.platform_name
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
+    parser.parse()
+
+    # check the kickstart test file exists
+    if not os.path.exists(parser.test_file_path):
+        print("kickstart test file is missing: {}".format(parser.test_file_path))
+        exit(1)
+
+    # check the requested platform folder exists:
+    if not os.path.exists(parser.platform_dir_path):
+        print("platform folder path not found: {}".format(parser.platform_dir_path))
+        exit(1)
+
+    # make sure the dynamic folder exists and is empty
+    if os.path.exists(DYNAMIC_FOLDER):
+        # cleanup the dynamic folder
+        shutil.rmtree(DYNAMIC_FOLDER)
+    else:
```

Comment:
I think you are removing the DYNAMIC FOLDER so it should be created in any case (not in the else block?